### PR TITLE
Add header alignment attribute

### LINKED ISSUE #102: Add header_align argument
The idea here is to be able to align the header separately from the values. For example, when I print floats I like the decimal points to align, and I get that with align='r'. But then the header names are all the way to the right, and you end up with

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;x
&nbsp;&nbsp;&nbsp;&nbsp;12346.657

whereas I'd like:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;x
&nbsp;&nbsp;&nbsp;&nbsp;12346.657

### PR DESCRIPTION
**Pull request by myheroyuki**
Fixes #102

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -295,6 +295,19 @@ gives:
...
+
+You can also change the alignment of the headers of the columns using the `header_align`
+attribute in the same way as the `align` attribute.
+
+```python
+x.header_align["City name"] = "l"
+print(x)
+```
+
+The valign attribute works in the same way as the other align attributes for controlling
```
Comment (hugovk) [inline README.md]:
```suggestion
The `valign` attribute works in the same way as the other align attributes for controlling
```

Comment (hugovk) [reply README.md]: And shall we mention the possible values for `valign`?

Comment (myheroyuki) [reply README.md]: Great idea! If there are any other changes you want me to add please let me know.

**Comment by hugovk** [comment]
Thank you!

**Comment by hugovk** [comment]
I only just released this in 3.4.0 and noticed a possible bug:

With this code:

```python
from prettytable import PrettyTable

headers = [
    "cycle",
    "release",
    "latest",
    "latest release",
    "support",
    "eol",
    "codename",
    "link",
]
data = [
    {
        "cycle": "22.04 LTS",
        "codename": "Jammy Jellyfish",
        "support": "2027-04-02",
        "eol": "2032-04-01",
        "latest": "22.04.1",
        "link": "https://wiki.ubuntu.com/JammyJellyfish/ReleaseNotes/",
        "release": "2022-04-21",
        "latest release": "2022-08-11",
    },
    {
        "cycle": "21.10",
        "codename": "Impish Indri",
        "support": "2022-07-31",
        "eol": "2022-07-31",
        "latest": "21.10",
        "link": "https://wiki.ubuntu.com/ImpishIndri/ReleaseNotes/",
        "release": "2021-10-14",
        "latest release": "2021-10-14",
    },
    {
        "cycle": "21.04",
        "codename": "Hirsute Hippo",
        "support": "2022-01-20",
        "eol": "2022-01-20",
        "latest": "21.04",
        "link": "https://wiki.ubuntu.com/HirsuteHippo/ReleaseNotes/",
        "release": "2021-04-22",
        "latest release": "2021-04-22",
    },
    {
        "cycle": "20.10",
        "codename": "Groovy Gorilla",
        "support": "2021-07-22",
        "eol": "2021-07-22",
        "latest": "20.10",
        "release": "2020-10-22",
        "latest release": "2020-10-22",
    },
    {
        "cycle": "20.04 LTS",
        "codename": "Focal Fossa",
        "support": "2025-04-02",
        "eol": "2030-04-01",
        "latest": "20.04.4",
        "release": "2020-04-23",
        "latest release": "2022-02-25",
    },
    {
        "cycle": "19.10",
        "codename": "Karmic Koala",
        "support": "2020-07-06",
        "eol": "2020-07-06",
        "latest": "19.10",
        "release": "2019-10-17",
        "latest release": "2019-10-17",
    },
    {
        "cycle": "18.04 LTS",
        "codename": "Bionic Beaver",
        "support": "2023-04-02",
        "eol": "2028-04-01",
        "latest": "18.04.6",
        "link": "https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes",
        "release": "2018-04-26",
        "latest release": "2021-09-17",
    },
    {
        "cycle": "16.04 LTS",
        "codename": "Xenial Xerus",
        "support": "2021-04-02",
        "eol": "2026-04-01",
        "latest": "16.04.7",
        "release": "2016-04-21",
        "latest release": "2020-08-13",
    },
    {
        "cycle": "14.04 LTS",
        "codename": "Trusty Tahr",
        "support": "2019-04-02",
        "eol": "2024-04-01",
        "latest": "14.04.6",
        "release": "2014-04-17",
        "latest release": "2019-03-07",
    },
]

x = PrettyTable()

for header in headers:
    col_data = [row[header] if header in row else "" for row in data]
    x.add_column(header, col_data)
    if header in ("cycle", "latest", "link"):
        x.align[header] = "l"

print(x)
```

Run with 3.3.0:

```console
$ pip install --quiet prettytable==3.3.0 && python 1.py
+-----------+------------+---------+----------------+------------+------------+-----------------+------------------------------------------------------+
| cycle     |  release   | latest  | latest release |  support   |    eol     |     codename    | link                                                 |
+-----------+------------+---------+----------------+------------+------------+-----------------+------------------------------------------------------+
| 22.04 LTS | 2022-04-21 | 22.04.1 |   2022-08-11   | 2027-04-02 | 2032-04-01 | Jammy Jellyfish | https://wiki.ubuntu.com/JammyJellyfish/ReleaseNotes/ |
| 21.10     | 2021-10-14 | 21.10   |   2021-10-14   | 2022-07-31 | 2022-07-31 |   Impish Indri  | https://wiki.ubuntu.com/ImpishIndri/ReleaseNotes/    |
| 21.04     | 2021-04-22 | 21.04   |   2021-04-22   | 2022-01-20 | 2022-01-20 |  Hirsute Hippo  | https://wiki.ubuntu.com/HirsuteHippo/ReleaseNotes/   |
| 20.10     | 2020-10-22 | 20.10   |   2020-10-22   | 2021-07-22 | 2021-07-22 |  Groovy Gorilla |                                                      |
| 20.04 LTS | 2020-04-23 | 20.04.4 |   2022-02-25   | 2025-04-02 | 2030-04-01 |   Focal Fossa   |                                                      |
| 19.10     | 2019-10-17 | 19.10   |   2019-10-17   | 2020-07-06 | 2020-07-06 |   Karmic Koala  |                                                      |
| 18.04 LTS | 2018-04-26 | 18.04.6 |   2021-09-17   | 2023-04-02 | 2028-04-01 |  Bionic Beaver  | https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes    |
| 16.04 LTS | 2016-04-21 | 16.04.7 |   2020-08-13   | 2021-04-02 | 2026-04-01 |   Xenial Xerus  |                                                      |
| 14.04 LTS | 2014-04-17 | 14.04.6 |   2019-03-07   | 2019-04-02 | 2024-04-01 |   Trusty Tahr   |                                                      |
+-----------+------------+---------+----------------+------------+------------+-----------------+------------------------------------------------------+
```

And with 3.4.0:

```console
pip install --quiet prettytable==3.4.0 && python 1.py
+-----------+------------+---------+----------------+------------+------------+-----------------+------------------------------------------------------+
|   cycle   |  release   |  latest | latest release |  support   |    eol     |     codename    |                         link                         |
+-----------+------------+---------+----------------+------------+------------+-----------------+------------------------------------------------------+
| 22.04 LTS | 2022-04-21 | 22.04.1 |   2022-08-11   | 2027-04-02 | 2032-04-01 | Jammy Jellyfish | https://wiki.ubuntu.com/JammyJellyfish/ReleaseNotes/ |
| 21.10     | 2021-10-14 | 21.10   |   2021-10-14   | 2022-07-31 | 2022-07-31 |   Impish Indri  | https://wiki.ubuntu.com/ImpishIndri/ReleaseNotes/    |
| 21.04     | 2021-04-22 | 21.04   |   2021-04-22   | 2022-01-20 | 2022-01-20 |  Hirsute Hippo  | https://wiki.ubuntu.com/HirsuteHippo/ReleaseNotes/   |
| 20.10     | 2020-10-22 | 20.10   |   2020-10-22   | 2021-07-22 | 2021-07-22 |  Groovy Gorilla |                                                      |
| 20.04 LTS | 2020-04-23 | 20.04.4 |   2022-02-25   | 2025-04-02 | 2030-04-01 |   Focal Fossa   |                                                      |
| 19.10     | 2019-10-17 | 19.10   |   2019-10-17   | 2020-07-06 | 2020-07-06 |   Karmic Koala  |                                                      |
| 18.04 LTS | 2018-04-26 | 18.04.6 |   2021-09-17   | 2023-04-02 | 2028-04-01 |  Bionic Beaver  | https://wiki.ubuntu.com/BionicBeaver/ReleaseNotes    |
| 16.04 LTS | 2016-04-21 | 16.04.7 |   2020-08-13   | 2021-04-02 | 2026-04-01 |   Xenial Xerus  |                                                      |
| 14.04 LTS | 2014-04-17 | 14.04.6 |   2019-03-07   | 2019-04-02 | 2024-04-01 |   Trusty Tahr   |                                                      |
+-----------+------------+---------+----------------+------------+------------+-----------------+------------------------------------------------------+
```

Notice the different header alignment:

```diff
< | cycle     |  release   | latest  | latest release |  support   |    eol     |     codename    | link                                                 |
---
> |   cycle   |  release   |  latest | latest release |  support   |    eol     |     codename    |                         link                         |
```

So the code had set:

```python
    if header in ("cycle", "latest", "link"):
        x.align[header] = "l"
```

In 3.3.0, the data _and_ header were left-aligned.

In 3.4.0, only the data was left-aligned, the header was centre-aligned.

Is this a bug?

Should `x.header_align` override `x.align`?

But if there's no `x.header_align`, should the `x.align` behaviour not change from 3.3.0?

**Comment by osfrickler** [comment]
We noticed this change in OpenStack testing, too, and I think that default behavior should not change in a minor release, so I'd prefer this to be treated as a bug and get fixed.

**Comment by SeanMooney** [comment]
this is the openstack nova bug https://bugs.launchpad.net/nova/+bug/1988482

currently, it's blocking our stable branch for the last release.
we are working around it by pinning for now but ideally, this would be addressed
so that the default behaviour did not change outside of a major release.

currently I'm not sure how else we could "fix" this in oepnstack as we cannot raise minium on already released branches and see cant simply update the test as that would then fail with the older release of prettytable.

**Comment by SeanMooney** [comment]
actully i think i understand how to work around this in OpenStack so i have proposed 
https://review.opendev.org/c/openstack/nova/+/855658 to manuly hardcode left alignment for now.

we can backport that as a workaround.

this is still likely a bug but we can maintain the old behavior by just setting both align and header_align

**Comment by hugovk** [comment]
I'll revert this and make a bug fix release.

Then this PR can be re-submitted, and we'll need extra tests to make sure things work with both `header_align` and `align`, with just one, and with neither.

Also the default `header_align="c"` in `add_column` needs checking:
```python
def add_column(self, fieldname, column, align="c", valign="t", header_align="c"):
```

**Comment by hugovk** [comment]
3.4.1 released:

https://pypi.org/project/prettytable/3.4.1/

**Comment by osfrickler** [comment]
Thx for the fast update, 3.4.1 works without failure for us

**Pull request merged**